### PR TITLE
Feature/1.2.2.11/read service requests

### PR DESCRIPTION
**Pull request by jnakaso**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- read list of service_requests from DB
- use PrimeReactTable to page serviceRequests

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by sharrief** [comment]
Added Sam and Dillon to avoid reviewing my own changes.
I decided to build on top of the work Jeff started here on the read services, because I need the appConstatnts part of the read services in order to make progress on the input form (to get the dropdown/radio selections to work)

**Comment by DillonOLeary** [comment]
Taking a look at this now

File: src/hooks/__mocks__/useAppConstants.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+const emptyRecord = {
+  id: '',
+  changed_at: '',
+  changed_by: '',
+  created_at: '',
+};
+
+export const data = {
```
Comment (DillonOLeary) [inline src/hooks/__mocks__/useAppConstants.ts:16]: I added app constants in this file: https://github.com/digitalaidseattle/seattle-humane-app/blob/sprint-3/supabase/seed.sql.

Should we delete that?

File: pages/clients/index.tsx [hunk 2]
```diff
@@ -59,38 +83,47 @@ function Clients() {
...
+      <span className="p-column-title">Description</span>
+      {rowData.description}
     </>
   );
 
-  const statusBodyTemplate = (rowData: ClientTicket) => (
+  const emailBodyTemplate = (rowData: ServiceRequestType) => (
+    <>
+      <span className="p-column-title">Description</span>
+      FIXME rowData.team_member.email
```
Comment (DillonOLeary) [inline pages/clients/index.tsx:110]: There are a few FIXMEs here. Are we ok merging this with those?

File: pages/animals/index.tsx [hunk 3]
```diff
@@ -22,10 +22,6 @@ function Animals() {
 
   const [clientDialog, setClientDialog] = useState(false);
 
-  useEffect(() => {
```
Comment (DillonOLeary) [inline pages/animals/index.tsx:25]: Why was this removed?

File: src/components/serviceRequest/ServiceInformationSection.tsx
Comment (DillonOLeary) [inline src/components/serviceRequest/ServiceInformationSection.tsx:1]: I got a lot of FIXMEs when I set this up locally
Screenshot 2024-05-09 at 3 10 18 PM

Comment (DillonOLeary) [reply src/components/serviceRequest/ServiceInformationSection.tsx:1]: Also the service request section was under the clients header

Comment (DillonOLeary) [reply src/components/serviceRequest/ServiceInformationSection.tsx:1]: There is no service request section, and the reports tab is returning a 404

File: src/services/ClientService.ts
Comment (DillonOLeary) [inline src/services/ClientService.ts:1]: I think this might be displaying under the wrong header

**Comment by tristan-88** [comment]
If you and Jeff are working on the read services, do I just need to create
the UI read request view for the service request tickets?

As far as tables goes there few fields /properties missing like status
ticket , urgency of service, previous service, and service source

Or this is irrelevant to service request?  If so my bad.

File: pages/animals/index.tsx (hunk 3 above)
Comment (sharrief) [reply pages/animals/index.tsx:25]: I don't have the context for it. Jeff removed this a month ago. Also removed the getTickets() function from the ClientService, which wasn't using the DB anyway, so perhaps it was just him cleaning up code. 

We'll get this and other pages working and fetching from the DB in the upcoming sprints.

File: pages/clients/index.tsx (hunk 2 above)
Comment (sharrief) [reply pages/clients/index.tsx:110]: For now I say yes. I'm trying to keep this PR small, so its just focused on the "Input Form" and not on these other pages. They were previously referring to fields that don't exist. So rather than fix them or totally delete this page, I we can leave it as is until we come back to it in future sprints.

File: src/components/serviceRequest/ServiceInformationSection.tsx
Comment (sharrief) [reply src/components/serviceRequest/ServiceInformationSection.tsx:1]: 1. I'm not getting FIXMEs in the client when running. We should have a call on this to see what the diff btw our envs are.
2. Yes, if you check sprint-3 (before this PR) you'll see that the table in this page is for service requests. I think Jeff was just updating this to reflect that. Perhaps its the menu item name that needs to be changed. (I'll make a note of this)
3. Reports is also returning a 404 on sprint-3 before this PR. I havent looked into that page much, but I don't recall if it was working previously.

File: src/hooks/__mocks__/useAppConstants.ts (hunk 1 above)
Comment (sharrief) [reply src/hooks/__mocks__/useAppConstants.ts:16]: No, what you have in the seed is great. This file is just mock data for unit tests. Our tests should/will run without a live DB connection. This data is used by the mock version of the supabase client during test runs. (note the folder location is `__mocks__`. see (Jest manual mocks)[https://jestjs.io/docs/manual-mocks])

File: src/services/ClientService.ts
Comment (sharrief) [reply src/services/ClientService.ts:1]: We are in the ClientService.ts file right now. But I think this comment is related to the one you made in ServiceInformationSection. There are issues with the headings and menu items. We will need to fix those in the future.

**Comment by sharrief** [comment]
> If you and Jeff are working on the read services, do I just need to create the UI read request view for the service request tickets? As far as tables goes there few fields /properties missing like status ticket , urgency of service, previous service, and service source Or this is irrelevant to service request? If so my bad.

@tristan-88 For the UI, we can re-use the existing ServiceRequestDialog. I would say it just needs a property that can put all of the fields in a read-only mode. Please look into that if you can.

As for the tables, yes we don't have all the fields we need in the schema yet, but let's not add them yet. For now, are only trying to get an E2E demo working. Once we have the E2E flow (create, save, view) then we will adjust it to make it correct regarding the data fields, etc. 

What's important right now is how we hook these components up to the database, since the patterns we establish now will be used in other places. So keep that frame of reference in mind. Try to come up with a general solution when you can so that we can easily apply it to other parts of the app later.

File: pages/clients/index.tsx (hunk 2 above)
Comment (sharrief) [reply pages/clients/index.tsx:110]: Correction, I've moved the input form changes into a different PR.

File: src/components/serviceRequest/ServiceInformationSection.tsx
Comment (sharrief) [reply src/components/serviceRequest/ServiceInformationSection.tsx:1]: Correction to 1. I do see FIXMEs now. These were added since there is no corresponding column in the db for these fields.

File: pages/animals/index.tsx (hunk 3 above)
Comment (DillonOLeary) [reply pages/animals/index.tsx:25]: Ok gotcha

File: src/services/ClientService.ts [hunk 4]
```diff
@@ -326,6 +334,32 @@ class ClientService {
       .eq('type', AppConstants.Category);
     return response.data;
   }
+
+  async getServiceRequestById(id: ServiceRequestType['id']): Promise<ServiceRequestType> {
```
Comment (DillonOLeary) [inline src/services/ClientService.ts:338]: It looks like these functions are not called. Is that the intention?

File: src/services/ClientService.ts [hunk 5]
```diff
@@ -326,6 +334,32 @@ class ClientService {
...
+  async getServiceRequests(opts: SearchOptions): Promise<ServiceRequestType[]> {
+    const { data, error } = await supabaseClient
+      .from('service_requests')
+      .select('*, team_members(*)')
+      .range(opts.first, opts.first + opts.rows);
+    if (error) throw new Error(error.message);
+    return data;
+  }
+
+  async getServiceRequestsTotalRecords(): Promise<number> {
```
Comment (DillonOLeary) [inline src/services/ClientService.ts:356]: Same with this one

**Review by DillonOLeary — COMMENTED** [review]
I'm still a little confused on this change, it's not clear what it changes in the app. I looked for the read service requests part of this, but they look like functions that are not called.

**Comment by sharrief** [comment]
Cleaned up this PR by moving the useAppConstants and useTeamMember hooks into their own PRs.
